cmd/create.go

```diff
@@ -89,20 +81,18 @@ EXAMPLES
 	cmd.Flags().BoolP("confirm", "c", false, "Prompt to confirm all options interactively (Env: $FUNC_CONFIRM)")
 
 	// Help Action
-	cmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
-		runCreateHelp(cmd, args, newClient)
-	})
+	cmd.SetHelpFunc(runCreateHelp)
 
 	// Run Action
 	cmd.RunE = func(cmd *cobra.Command, args []string) error {
 		return runCreate(cmd, args, newClient)
 	}
 
-	// Tab Completion
-	if err := cmd.RegisterFlagCompletionFunc("language", newRuntimeCompletionFunc(newClient)); err != nil {
+	// Tab completion
+	if err := cmd.RegisterFlagCompletionFunc("language", newRuntimeCompletionFunc()); err != nil {
```

Comment:
Here and some other places there is removed propagation of `newClient`, instead the `cmd.NewClient()` is called directly.
I believe this shouldn't be done this way. It affect potential testability.

Reply:
Thanks @matejvasek !

"potential testability" is not actual tests, and as such I see no reason to pass an argument which is not being used.

Reply:
The ClientFactory is now passed through in all situations

Reply:
I think you might have misses on test file.

Reply:
I had `TestCreateConfig_RepositoriesPath` specifically in mind here. Because of no propagation it had to be rewritten --accessing/testing internal details (`newCreateConfig`). With propagation it can be reverted I think.

Reply:
>I had TestCreateConfig_RepositoriesPath specifically in mind here. Because of no propagation it had to be rewritten -- accessing/testing internal details (newCreateConfig). With propagation it can be reverted I think.

or maybe not.

Reply:
> I think you might have misses on test file.

It has been un-missed

Reply:
> accessing/testing internal details (`newCreateConfig`)

Yes, I think this is probably just fine, because the use of XDG_CONFIG_HOME to derive the config home directory is an implementation detail of the `func` package's config, which is tested in `config_test.go`.  The `cmd` package is only utilizing this feature in order to pre-populate a flag which ends up in the command's config object.  The responsibility of the cmd package is to ensure this is plumbed through from flag to config.  So each is testing _their part_ of the puzzle; not each-others :)